packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -1,80 +1,65 @@
+    private ecdsaValidator: KernelValidator<"ECDSAValidator">;
+    kernelClient: KernelAccountClient;
+    account: KernelSmartAccount;
     chain: B;
 
-    constructor(provider: ZeroDevEthersProvider<"ECDSA">, crossmintService: CrossmintWalletService, chain: B) {
-        super(provider, crossmintService);
+    constructor(
+        account: KernelSmartAccount,
+        crossmintService: CrossmintWalletService,
+        chain: B,
+        publicClient: PublicClient,
+        ecdsaValidator: KernelValidator<"ECDSAValidator">
+    ) {
         this.chain = chain;
-    }
-
-    async getSigner<Type extends SignerType = SignerType>(type: Type): Promise<SignerMap[Type]> {
-        switch (type) {
-            case "ethers":
-                return this.provider.getSigner() as any;
-            case "viem": {
-                const customeip1193Provider = new ZeroDevEip1193Bridge(this.provider.getAccountSigner(), this.provider);
-                const customTransport = custom({
-                    async request({ method, params }) {
-                        return customeip1193Provider.send(method, params);
-                    },
+        this.account = account;
+        this.crossmintService = crossmintService;
+        this.publicClient = publicClient;
+        this.ecdsaValidator = ecdsaValidator;
+
+        this.kernelClient = createKernelAccountClient({
+            account,
+            chain: getViemNetwork(chain as EVMBlockchainIncludingTestnet),
+            transport: http(getBundlerRPC(chain)),
+            sponsorUserOperation: async ({ userOperation }): Promise<UserOperation> => {
+                const paymasterClient = createZeroDevPaymasterClient({
+                    chain: getViemNetwork(chain as EVMBlockchainIncludingTestnet),
+                    transport: http(getPaymasterRPC(chain)),
                 });
-                const walletClient = createWalletClient({
-                    account: await this.getAddress(),
-                    chain: getViemNetwork(this.chain),
-                    transport: customTransport,
-                }).extend(publicActions);
-                return walletClient as any;
-            }
-            default:
-                logError("[GET_SIGNER] - ERROR", {
-                    service: SCW_SERVICE,
-                    error: errorToJSON("Invalid signer type"),
+                return paymasterClient.sponsorUserOperation({
+                    userOperation,
                 });
-                throw new Error("Invalid signer type");
-        }
-    }
-
-    async verifyMessage(message: string, signature: string) {
-        return verifyMessage({
-            provider: this.provider,
-            signer: await this.getAddress(),
-            message,
-            signature,
-        });
+            },
+        }) as KernelAccountClient;
     }
 
-    setSessionKeySignerAddress(sessionKeySignerAddress: string) {
+    setSessionKeySignerAddress(sessionKeySignerAddress: Hex) {
```

Comment:
same thought–should a client call this directly? Or should we mark as `private`?

Reply:
same thought–should a client call this directly? Or should we mark as `private`?
```suggestion
    private setSessionKeySignerAddress(sessionKeySignerAddress: Hex) {
```